tgui/packages/tgui/interfaces/ListInput.js

```diff
@@ -0,0 +1,235 @@
+                  }
+                }
+
+                if (foundValue) {
+                  setLastCharCode(charCode);
+                  setSelectedButton(foundValue);
+                  document.getElementById(foundValue).focus();
+                }
+              }}
+              buttons={
+                <Button
+                  icon="search"
+                  color="transparent"
+                  selected={showSearchBar}
+                  tooltip="Search..."
+                  tooltipPosition="left"
+                  onClick={() => {
+                    setShowSearchBar(!showSearchBar);
+                    setDisplayedArray(buttons);
+                  }}
+                  compact
+                />
+              }
+            >
+              <Flex wrap="wrap">
+                {displayedArray.map((button) => (
+                  <Flex.Item key={button} basis="100%">
+                    <Button
+                      color="transparent"
+                      content={button}
+                      id={button}
+                      width="100%"
+                      selected={selectedButton === button}
+                      onClick={() => {
+                        if (selectedButton === button) {
+                          act('choose', { choice: button });
+                        } else {
+                          setSelectedButton(button);
+                        }
+                        setLastCharCode(null);
+                      }}
+                    />
+                  </Flex.Item>
+                ))}
+              </Flex>
+            </Section>
+          </Flex.Item>
+          {showSearchBar && (
+            <Flex.Item basis={2.5}>
+              <Input
+                width="100%"
+                autoFocus
+                onInput={(e, value) =>
+                  setDisplayedArray(
+                    buttons.filter(
+                      (val) =>
+                        val.toLowerCase().search(value.toLowerCase()) !== -1
+                    )
+                  )
+                }
+              />
+            </Flex.Item>
+          )}
+          <Flex.Item>
+            <Flex textAlign="center">
+              <Flex.Item grow={1} basis={0} ml={1} mx="5px">
+                <Button
+                  fluid
+                  color="good"
+                  content="Confirm"
+                  disabled={selectedButton === null}
+                  onClick={() => act('choose', { choice: selectedButton })}
+                />
+              </Flex.Item>
+              <Flex.Item grow={1} basis={0} mr={1} mx="5px">
+                <Button
+                  fluid
+                  color="bad"
+                  content="Cancel"
+                  onClick={() => act('cancel')}
+                />
+              </Flex.Item>
+            </Flex>
+          </Flex.Item>
+        </Flex>
+      </Window.Content>
+    </Window>
+  );
+};
+
+export const Loader = (props) => {
```

Comment:
We should keep this as a const for now imo, since we don't use it anywhere else
```suggestion
const Loader = (props) => {
```